crates/bevy_render/src/mesh/primitives/circle_iterator.rs

```diff
@@ -0,0 +1,80 @@
+use bevy_math::Vec2;
+use std::f32::consts::TAU;
+
+#[derive(Debug, Clone)]
+pub(crate) struct CircleIterator {
+    count: usize,
+    theta: f32,
+    wrap: bool,
+}
+
+impl CircleIterator {
+    pub(crate) fn new(count: usize, wrap: bool) -> CircleIterator {
+        if wrap {
+            Self {
+                count: count + 1,
+                theta: -TAU / (count as f32),
+                wrap,
+            }
+        } else {
+            Self {
+                count,
+                theta: -TAU / (count as f32),
+                wrap,
+            }
+        }
+    }
+}
+impl Iterator for CircleIterator {
+    type Item = Vec2;
```

Comment:
Just opening a discussion here:

Thoughts on using `Vec2` here? Having the resulting values accessed by `.x` and `.y` seems sort of confusing.

Maybe `(f32, f32)` or perhaps `SinCos { sin: f32, cos: f32 }`?

Reply:
`Vec2` was chosen because the first alternative to sin/cos calls that I wanted to investigate was application of a `Mat2` for rotation, which would get generated on the initialization of the iterator - which could then be subsequently cloned for creation of identical sets of unit circle points. 

If this method does not yield favorable results, I agree that something else should be used (and if it does, it would still make sense to have it return something with labels that are more intuitive than `.x`, `.y`).